pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -1259,18 +1245,54 @@ func readLatestCheckpointFile(
 	// First try reading from the progress directory. If the backup is from
 	// an older version, it may not exist there yet so try reading
 	// in the base directory if the first attempt fails.
-	r, err := exportStore.ReadFile(ctx, backupProgressDirectory+"/"+filename)
-	if err != nil {
+	var checkpoint string
+	var checkpointFound bool
+	var r ioctx.ReadCloserCtx
+	var err error
+
+	// HTTP doesn't support listing so only try reading the non-timestamped
+	// version of the filename in the progress directory.
+	if exportStore.Conf().Provider == roachpb.ExternalStorageProvider_http {
+		r, err = exportStore.ReadFile(ctx, backupProgressDirectory+"/"+filename)
+		if err == nil {
+			return r, err
+		}
+	} else {
+		// We name files such that the most recent checkpoint will always
+		// be at the top, so just grab the first filename. File we're looking
+		// for can be CHECKPOINT or CHECKSUM, so we want to pass in filename.
+		err = exportStore.List(ctx, backupProgressDirectory+"/"+filename, "", func(p string) error {
+			p = strings.TrimPrefix(p, "/")
+			checkpoint = p
+			checkpointFound = true
+			return nil
+		}, /*limit*/ 1)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// The checkpoint file couldn't be found in the progress directory,
+	// try the base directory instead.
+	if !checkpointFound {
 		r, err = exportStore.ReadFile(ctx, filename)
 		if err != nil {
 			return nil, errors.Wrapf(err, "%s could not be read in the base or progress directory", filename)
 		}
+		return r, nil
+	}
+	// List trims off the filename, so prepend it back on.
+	checkpoint = filename + checkpoint
+	r, err = exportStore.ReadFile(ctx, backupProgressDirectory+"/"+checkpoint)
+	if err != nil {
+		return nil, err
 	}
 	return r, nil
 }
 
-// writeNewCheckpointFile writes a new BACKUP-CHECKPOINT file to both the base
-// directory and latest-history directory, depending on cluster version.
+// writeNewCheckpointFile writes a new BACKUP-CHECKPOINT file to either
```

Comment:
nit: writeNewCheckpointFile writes a new checkpoint (or checksum) file to the appropriate directory depending on the cluster version.